src/app/_components/Search/RestultItem.tsx

```diff
@@ -0,0 +1,243 @@
+import { Flex, FlexProps, Icon } from '@chakra-ui/react';
+import { ArrowRight, KeyReturn } from '@phosphor-icons/react';
+import * as React from 'react';
+
+import { ContractCallTxs, ContractDeployTxs, TokenTransferTxs } from '../../../common/types/tx';
+import {
+  getContractName,
+  microToStacksFormatted,
+  truncateMiddle,
+} from '../../../common/utils/utils';
+import { Text } from '../../../ui/Text';
+import { TextLink } from '../../../ui/TextLink';
+import StxIcon from '../../../ui/icons/StxIcon';
+import { SearchLink } from './SearchLink';
+import { TxTag } from './TxTag';
+
+function ResultItemWrapper({ children, ...props }: FlexProps) {
+  return (
+    <Flex
+      background={'surfacePrimary'}
+      borderRadius={'md'}
+      px={3.5}
+      py={3}
+      alignItems={'center'}
+      gap={2}
+      justifyContent={'space-between'}
+      {...props}
+    >
+      {children}
+    </Flex>
+  );
+}
+
+function ResultItemIcon({ type }: { type?: 'arrow' | 'enter' | undefined }) {
+  if (type === 'enter') {
+    return (
+      <Flex alignItems={'center'} gap={1.5} flex={'0 0 auto'}>
+        <Icon h={3.5} w={3.5} color={'iconSecondary'}>
+          <KeyReturn />
+        </Icon>
+        <Text fontSize={'xs'} color={'textSecondary'}>
+          Enter
+        </Text>
+      </Flex>
+    );
+  }
+  if (type === 'arrow') {
+    return (
+      <Icon h={4} w={4} color={'iconSecondary'}>
+        <ArrowRight />
+      </Icon>
+    );
+  }
+  return null;
+}
+
+export function ResultItem({ value }: { value: string }) {
+  return (
+    <ResultItemWrapper>
```

Comment:
Every ResultItem in this file basically looks the same to me. It looks like:
<ResultItemWrapper>
      <Flex gap={4} flex={'1 1 auto'} minWidth={0}>
        <SearchLink href={'#'}>
            some text
        </SearchLink>
        <SomeCustomComponentPieces />
        </Flex>
         <ResultItemIcon type={'arrow'} />
    </ResultItemWrapper>
    Can you see if it's possible to write one component for all of these and that just takes the tx type and uses it to render the specific  <SomeCustomComponentPieces /> it needs?